src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentStack.cs

```diff
@@ -384,7 +384,7 @@ private void PushCore(Node head, Node tail)
             // Keep trying to CAS the existing head with the new node until we succeed.
             do
             {
-                spin.SpinOnce();
+                spin.SpinOnce(sleep1Threshold: -1);
```

Comment:
I'm a little surprised the SpinOnce call here is valuable.  This is one of the ones you tried removing and saw a regression as a result?

Reply:
Yea I tried removing this one and the one in Pop, simplifying the one in pop to remove the extra backoff, so far those things didn't pan out, even the extra backoff in Pop seems to be (marginally) better than just one SpinOnce().